sherlock/db/migrations/000062_environment_v3.up.sql

```diff
@@ -0,0 +1,54 @@
+alter table environments
+    add constraint name_valid
+        check (name != ''  and name similar to '[a-z0-9]([-a-z0-9]*[a-z0-9])?');
+
+alter table environments
+    add constraint owner_id_present
+        check (owner_id is not null or
+               (legacy_owner is not null and legacy_owner != ''));
+
+alter table environments
+    add constraint lifecycle_valid
+        check ((lifecycle = 'template' and template_environment_id is null) or
+               (lifecycle = 'dynamic' and template_environment_id is not null) or
+               (lifecycle = 'static' and
+                base != '' and
+                default_cluster_id is not null and
+                requires_suitability is not null));
+
+alter table environments
+    add constraint default_namespace_present
+        check (default_namespace != '');
```

Comment:
unfortunately I think SQL null passes this check. I know we probably wouldn't be able to insert null through Gorm in most cases, but I think we should write these sorts of checks (here, line 29, line 15) like you did on line 25. I don't think you need to change the go tests at all because yeah you'd have a hard time inserting null